### PR TITLE
docs: explain deployment.environment impact on service identity

### PR DESCRIPTION
**Pull request by bripkens**
# Why

To clarify a likely misconception about service identity in the case of multiple environments.

# What

Explain that `deployment.environment` does not impact service identity.

# References

Fixes https://github.com/open-telemetry/semantic-conventions/issues/119

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: model/resource/deployment_environment.yaml [hunk 1]
```diff
@@ -10,4 +10,12 @@ groups:
         brief: >
           Name of the [deployment environment](https://wikipedia.org/wiki/Deployment_environment)
           (aka deployment tier).
+        note: >
+          `deployment.environment` does not affect the uniqueness constraints defined through
```
Comment (joaopgrassi) [inline model/resource/deployment_environment.yaml:14]: Hi @bripkens 
You need to re-generate the markdowns after changing the YAML here. See here https://github.com/open-telemetry/semantic-conventions/blob/main/model/README.md#generating-markdown

Comment (bripkens) [reply model/resource/deployment_environment.yaml:14]: Thank you @joaopgrassi, done

**Review by Oberon00 — APPROVED** [review]

**Comment by Oberon00** [comment]
I think this is a good description of the current state. Maybe we should also add a note that service.namespace could be used to distinguish the services. A more major change would be to make the environment part of the service identity, although IMHO we should then rename the attribute to service.environment (or service.deployment_environment?)

File: model/resource/deployment_environment.yaml [hunk 2]
```diff
@@ -10,4 +10,12 @@ groups:
...
+          the `service.namespace`, `service.name` and `service.instance.id` resource attributes.
+          This implies that resources carrying the following attribute combinations MUST be
+          considered to be identifying the same service:
+
```
Comment (Oberon00) [inline model/resource/deployment_environment.yaml:18]:
```suggestion

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ release.
 
 ## Unreleased
 
+- Explain `deployment.environment` impact on service identity. ([#481](https://github.com/open-telemetry/semantic-conventions/pull/481))
```
Comment (joaopgrassi) [inline CHANGELOG.md:10]: I think this should be moved down to the `Fixes` section.

**Review by joaopgrassi — APPROVED** [review]

**Review by arminru — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (bripkens) [reply CHANGELOG.md]: done @joaopgrassi

**Review by AlexanderWert — APPROVED** [review]

**Pull request merged**